### PR TITLE
Stabilize rounding to increment methods for `FixedDecimal`

### PR DESCRIPTION
**Pull request by jedel1043**
Follow up of #4571.

### CODE REVIEW

**Comment by jedel1043** [comment]
The failing action seems to be sporadic; I ran it locally without problems.

**Comment by sffc** [comment]
Note: No binsize diff.

Main:

```
-rwxr-xr-x 1 root root    24248 Feb  5 23:27 optim4.elf
-rwxr-xr-x 1 root root    13520 Feb  5 23:27 optim5.elf
```

PR:

```
-rwxr-xr-x 1 root root    24256 Feb  5 14:00 optim4.elf
-rwxr-xr-x 1 root root    13520 Feb  5 14:00 optim5.elf
```

**Comment by sffc** [comment]
https://github.com/unicode-org/icu4x/blob/main/docs/tutorials/c-tiny/fixeddecimal/test.c doesn't include calls to rounding functions. I think we should probably add that.

**Comment by sffc** [comment]
On main:

```
rwxr-xr-x 1 root root    25568 Feb  6 16:43 optim4.elf
-rwxr-xr-x 1 root root    14464 Feb  6 16:43 optim5.elf
```

On this branch:

```
-rwxr-xr-x 1 root root    29112 Feb  6 01:16 optim4.elf
-rwxr-xr-x 1 root root    17520 Feb  6 01:16 optim5.elf
```

🐖

**Comment by sffc** [comment]
@jedel1043 I think we can use generics to solve this and stay DRY. Sandbox example:

https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=c9c7c4788888d36fc3e1b37e7b9dd36a

Build it as WASM and check the output. Note that `trunc` and `trunc_to_increment` compile to different code, with `trunc` being substantially smaller than `trunc_to_increment`.

This will cause duplicated code between the increment and non-increment functions, but I think this is okay because if you care about code size you should just use either one or the other, and we know that many users don't need increments so we don't want to penalize those callers.

**Comment by jedel1043** [comment]
Did a size comparison of all possible solutions:

main
```bash
-rwxr-xr-x 1 jedel users    25376 Feb  6 23:19 optim4.elf
-rwxr-xr-x 1 jedel users    14728 Feb  6 23:19 optim5.elf
```
RoundingIncrement
```bash
-rwxr-xr-x 1 jedel users    28768 Feb  6 23:27 optim4.elf
-rwxr-xr-x 1 jedel users    17656 Feb  6 23:27 optim5.elf
```
IncrementLike
```bash
-rwxr-xr-x 1 jedel users    26128 Feb  7 00:25 optim4.elf
-rwxr-xr-x 1 jedel users    15000 Feb  7 00:25 optim5.elf
```
IncrementLike + #[inline(never)]
```bash
-rwxr-xr-x 1 jedel users    25648 Feb  7 00:39 optim4.elf
-rwxr-xr-x 1 jedel users    14920 Feb  7 00:39 optim5.elf
```

**Review by sffc — APPROVED** [review]
Thanks for making this change! This looks good to me now to land. After it lands we can play around a little more with the `#[inline]` attributes to measure the impact.

I'm going to copy @markusicu on this PR in order to grant the stabilization review.

File: ffi/capi/src/fixed_decimal.rs
Comment (robertbastian) [inline ffi/capi/src/fixed_decimal.rs:44]: issue: this is a released stable API, we cannot rename it.

Comment (jedel1043) [reply ffi/capi/src/fixed_decimal.rs:44]: Was thinking of undoing the rename and adding a comment about renaming it for 2.0, but maybe it would be better to `#[deprecate]` the old name instead. I think that would be harder to miss for the 2.0 release. What do you think?

Comment (robertbastian) [reply ffi/capi/src/fixed_decimal.rs:44]: leave a comment. we're going to remove all the `ICU4X` prefixes in 2.0, so we'll touch this code.

**Review by robertbastian — APPROVED** [review]
Please wait for @markusicu review as well

**Review by sffc — APPROVED** [review]

**Comment by sffc** [comment]
Notes from the API Review:

https://docs.google.com/document/d/1gKZCaiTG2eeyxrIKUF2tf6PIsaPCIBaUVRXD-JDh6dI/edit#heading=h.7pm5mstzwo40

@markusicu left numerous suggestions about documentation and generally the FixedDecimal API shape which we can fix in the 2.0 timeframe. I will file issues about those. However, @markusicu was happy with the shape and naming of `RoundingIncrement` and `*_to_increment`, so we have the i18n Correctness approval to land the new API.

Thanks @jedel1043!

**Pull request merged**
